File: src/test/resources/InorderTraversal.java
import java.util.*;
import org.algohub.engine.type.*;

public class Solution {

    public int[] inorderTraversal(TreeNode root) {
        List<Integer> items = new ArrayList<>();
        inorder(root, items);
        return items.stream().mapToInt(i -> i).toArray();
    }

    private void inorder(TreeNode root, List<Integer> items) {
        if (root != null) {
            inorder(root.left, items);
            items.add(root.data);
            inorder(root.right, items);
        }
    }
}

File: src/test/resources/InorderTraversal.rb
class Solution
  def inorderTraversal(root)
    items = []
    inorder(root, items)
    items.map(&:to_i)
  end

  def inorder(root, items)
    if root
      inorder(root.left, items)
      items << root.data
      inorder(root.right, items)
    end
  end
end